auslib/web/public/api.py

```diff
@@ -0,0 +1,157 @@
+import urllib
+import re
+
+from flask import make_response, request, current_app as app
+
+from auslib.AUS import AUS
+from auslib.global_state import dbo
+
+import logging
+
+AUS = AUS()
+LOG = logging.getLogger(__name__)
+
+
+def getHeaderArchitecture(buildTarget, ua):
+    if buildTarget.startswith('Darwin'):
+        if ua and 'PPC' in ua:
+            return 'PPC'
+        else:
+            return 'Intel'
+    else:
+        return 'Intel'
+
+
+def removeAvastBrokenness(locale):
+    # Some versions of Avast make requests and blindly append "?avast=1" to
+    # them, which breaks query string parsing if ?force=1 is already
+    # there. Because we're nice people we'll fix it up.
+    qs = request.environ.get("QUERY_STRING", "")
+    if "force" in qs and "avast" in qs:
+        qs = qs.replace("?avast=1", "&avast=1")
+        qs = qs.replace("%3Favast=1", "&avast=1")
+        request.environ["QUERY_STRING"] = qs
+    # Some versions of Avast have a bug in them that prepends "x86 "
+    # to the locale. We need to make sure we handle this case correctly
+    # so that these people can keep up to date.
+    return locale.replace("x86 ", "")
+
+
+def getQueryFromURL(url):
+    query = url.copy()
+    query["locale"] = removeAvastBrokenness(query["locale"])
+    query['osVersion'] = urllib.unquote(query['osVersion'])
+    ua = request.headers.get('User-Agent')
+    query['headerArchitecture'] = getHeaderArchitecture(query['buildTarget'], ua)
+    query['force'] = (int(request.args.get('force', 0)) == 1)
+    return query
+
+
+def get_update_blob_1(**url):
+    return get_update_blob(**url)
+
+
+def get_update_blob_2(**url):
+    return get_update_blob(**url)
+
+
+def get_update_blob_3(**url):
+    return get_update_blob(**url)
+
+
+def get_update_blob_3_esrpre(**url):
+    return get_update_blob(**url)
+
+
+def get_update_blob_4(**url):
+    return get_update_blob(**url)
+
+
+def get_update_blob_5(**url):
+    return get_update_blob(**url)
+
+
+def get_update_blob_6(**url):
+    return get_update_blob(**url)
```

Comment:
Nice find there. Were these tests passing before you started re-using operation ids?

This is very strange, it looks like you're mocking correctly, yet the function still runs. It makes me wonder if connexion is creating a copy of the function under the hood, or something. If so, this might be very difficult to deal with :(.